File: new_server/vueblog-common/src/main/java/cn/smilex/vueblog/common/entity/blog/SearchBlog.java
package cn.smilex.vueblog.common.entity.blog;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author smilex
 * @date 2022/12/10/18:25
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class SearchBlog {
    private Long id;
    private String title;
    private String description;
    private String content;

    public static SearchBlog fromRequestBlog(RequestBlog requestBlog) {
        SearchBlog searchBlog = new SearchBlog();
        searchBlog.setId(requestBlog.getId());
        searchBlog.setTitle(requestBlog.getTitle());
        searchBlog.setDescription(requestBlog.getDescription());
        searchBlog.setContent(requestBlog.getContent());
        return searchBlog;
    }
}

File: new_server/vueblog-common/src/main/java/cn/smilex/vueblog/common/entity/blog/SearchBlog.kt
package cn.smilex.vueblog.common.entity.blog

/**
 * @author smilex
 * @date 2023/12/4 22:00:52
 */
data class SearchBlog(
    var id: Long? = null,
    var title: String? = null,
    var description: String? = null,
    var content: String? = null
) {
    companion object {
        @JvmStatic
        fun fromRequestBlog(requestBlog: RequestBlog): SearchBlog {
            val searchBlog = SearchBlog()

            searchBlog.id = requestBlog.id
            searchBlog.title = requestBlog.title
            searchBlog.description = requestBlog.description
            searchBlog.content = requestBlog.content

            return searchBlog
        }

        @JvmStatic
        fun fromBlog(blog: Blog): SearchBlog {
            return SearchBlog(
                blog.id,
                blog.title,
                blog.description,
                blog.content
            )
        }
    }
}